interface-impl-amf/src/test/java/org/mule/amf/impl/BaseUriParametersResolutionTestCase.java

```diff
@@ -0,0 +1,160 @@
+    assertEquals(1, resolvedUriParameters.size());
+    assertNotNull(resolvedUriParameters.get("apiDomain"));
+    resolvedUriParameters = teamsResource.getAction(POST.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(1, resolvedUriParameters.size());
+    assertNotNull(resolvedUriParameters.get("apiDomain"));
+
+    Resource teamsResourceId = teamsResource.getResources().get("/{teamId}");
+    resolvedUriParameters = teamsResourceId.getAction(GET.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(1, resolvedUriParameters.size());
+    assertNotNull(resolvedUriParameters.get("teamId"));
+    resolvedUriParameters = teamsResourceId.getAction(PUT.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(2, resolvedUriParameters.size());
+    assertNotNull(resolvedUriParameters.get("teamId"));
+    assertNotNull(resolvedUriParameters.get("apiDomain"));
+    resolvedUriParameters = teamsResourceId.getAction(DELETE.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(1, resolvedUriParameters.size());
+    assertNotNull(resolvedUriParameters.get("teamId"));
+
+    Resource positionsResource = resources.get("/positions");
+    resolvedUriParameters = positionsResource.getAction(GET.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(0, resolvedUriParameters.size());
+
+    Resource fixtureResource = resources.get("/fixture");
+    resolvedUriParameters = fixtureResource.getAction(GET.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(0, resolvedUriParameters.size());
+
+    Resource teamsMatchResource = fixtureResource.getResources().get("/{homeTeamId}/{awayTeamId}");
+    resolvedUriParameters = teamsMatchResource.getAction(GET.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(2, resolvedUriParameters.size());
+    assertNotNull(resolvedUriParameters.get("homeTeamId"));
+    assertNotNull(resolvedUriParameters.get("awayTeamId"));
+    resolvedUriParameters = teamsMatchResource.getAction(PUT.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(2, resolvedUriParameters.size());
+    assertNotNull(resolvedUriParameters.get("homeTeamId"));
+    assertNotNull(resolvedUriParameters.get("awayTeamId"));
+
+    Resource badgeResource = resources.get("/badge");
+    resolvedUriParameters = badgeResource.getAction(PUT.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(1, resolvedUriParameters.size());
+    assertNotNull(resolvedUriParameters.get("apiDomain"));
+    resolvedUriParameters = badgeResource.getAction(GET.name()).getResolvedUriParameters();
+    assertNotNull(resolvedUriParameters);
+    assertEquals(0, resolvedUriParameters.size());
```

Comment:
Would nice to have this on a separate test, to clearly identify the issue.

Reply:
Done